subsys/bluetooth/mesh/Kconfig.models

```diff
@@ -653,11 +653,11 @@ menuconfig BT_MESH_SCENE_SRV
 
 config BT_MESH_SCENES_MAX
 	int "Max number of scenes"
-	default 4
-	range 1 65535
+	default 16
```

Comment:
Why did we change default to 16 instead of 4? I think we should keep it as 4 as default.

Reply:
Because the specification says that the scene register "is a 16-element array". It is debatable if having this configurable is even allowed by the spec, the least we should do is change the default to what the spec specifies.

Reply:
Maybe we have mix the total number of scenes together with how many a single scene server can store. I see the point that we maybe just should remove this config entry, and just define it. It will increase the flash usage. Doc need to be changed too.

Reply:
A single-scene server shall always be able to store up to 16 scenes. The actual numerical value stored in that scene entry can range from 0 to 65535.

Reply:
I think we should even remove the "prompt" here and simply set the value to 16, to indicate that this is a mandatory value, and still allow vendor models to change Kconfig file if they want to create Enhanced Scene Server vendor model that supports many scene register entries.

Reply:
If this is the intention, perhaps range should be extended? I think vendor models can't set it out of range, and with no prompt we don't need the range for application development. To be honest, I'd actually like to just remove this config entirely, makes for less configurations to maintain.